pkg/upgrade/upgrades/update_invalid_column_ids_in_sequence_back_references.go

```diff
@@ -0,0 +1,127 @@
+// Copyright 2022 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package upgrades
+
+import (
+	"context"
+	"sort"
+
+	"github.com/cockroachdb/cockroach/pkg/clusterversion"
+	"github.com/cockroachdb/cockroach/pkg/jobs"
+	"github.com/cockroachdb/cockroach/pkg/kv"
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog"
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog/descpb"
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog/descs"
+	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
+	"github.com/cockroachdb/cockroach/pkg/sql/sqlutil"
+	"github.com/cockroachdb/cockroach/pkg/upgrade"
+	"github.com/cockroachdb/errors"
+)
+
+func updateInvalidColumnIDsInSequenceBackReferences(
+	ctx context.Context, _ clusterversion.ClusterVersion, d upgrade.TenantDeps, _ *jobs.Job,
+) error {
+	// Upgrade each sequence, one at a time, until we exhaust all of them.
```

Comment:
I suspect this will be fine, but if you wanted to be conservative, you could use a new transaction for each sequence you need to upgrade. Right now, you're going to have a long-running transaction holding locks on the descriptor table. I don't know if that's desirable. 

I'm okay with this migration taking `O(sequences to be repaired)` transactions. You'll need to return some indication that `maybeUpdateInvalidColumnIdsInSequenceBackReferencesBestEffort` did indeed do something.

Reply:
I've changed it back to one sequence per transaction. I agreed that it's not desirable to have potentially long running transactions holding up locks on a table.

One question though pertaining to the new interface though is:
If I have a collectionFactory, I can now do `collectionFactory.TxnWithExecutor(..., desc.Collection, sqlutil.InternalExecutor...)` where I can have access to both a `desc.Collection` and a `InternalExecutor`, and the internal executor is tied to the desc collection.

But now, `tenantDeps` has both a `collectionFactory` and a `internalExecutor`. I'm using `tenantDeps.ie` to retrieve the next sequence ID to upgrade first, and later I use `tenantDeps.CollectionFactory.TxnWithExecutor(....,  _ sqlutil.InternalExecutor)` for the actual upgrade logic without using the provided internal executor. 

Will there be any correctness concerns, similar to what Jane said [here](https://github.com/cockroachdb/cockroach/pull/86255#discussion_r951566032)?